File: typescript/src/tools/my_usage.ts
import { apiRequest } from "../client.js";
import { mcpError } from "../errors.js";
import type { RawPersonalStats } from "../schema.js";
import { myUsageInputSchema } from "../types.js";

export const name = "toolrate_my_usage";

export const config = {
  title: "Check your ToolRate API key quota and usage",
  description: "Returns your tier, daily/monthly limit, calls used, and remaining quota. Use this to self-throttle before burning your free tier.",
  inputSchema: myUsageInputSchema,
};

export async function handler(_input: Record<string, never>) {
  try {
    const { body, rateLimit } = await apiRequest<RawPersonalStats>({
      method: "GET",
      path: "/v1/stats/me",
    });

    const result = {
      ...body,
      ...(rateLimit ? { _meta: { rate_limit: rateLimit } } : {}),
    };

    return {
      content: [{ type: "text" as const, text: JSON.stringify(result, null, 2) }],
    };
  } catch (err) {
    return mcpError(err);
  }
}

File: python/src/toolrate_mcp/tools/my_usage.py
"""``toolrate_my_usage`` — check your own quota."""
from __future__ import annotations

import json
from typing import Any

from ..client import api_request
from ..errors import raise_tool_error

NAME = "toolrate_my_usage"

DESCRIPTION = (
    "Returns your tier, daily/monthly limit, calls used, and remaining quota. "
    "Use this to self-throttle before burning your free tier."
)


async def handler() -> str:
    try:
        body, rate_limit = await api_request(method="GET", path="/v1/stats/me")
    except Exception as e:
        raise_tool_error(e)

    result: dict[str, Any] = {**body}
    if rate_limit is not None:
        result["_meta"] = {"rate_limit": rate_limit}
    return json.dumps(result, indent=2)
